### PR TITLE
Added in unit tests for Synchronizer

### PR DESCRIPTION
**Pull request by SeanCondon**

### CODE REVIEW

File: pkg/southbound/synchronizer/factory_test.go [hunk 1]
```diff
@@ -0,0 +1,140 @@
...
+		assert.Error(t, resp.Error(),
+			"could not create a gNMI client: Dialer(1.2.3.4:11161, 500ms): context deadline exceeded", "after gRPC timeout")
+		break
+	}
+
+	/*****************************************************************
+	 * Now it should have cleaned up after itself
+	 *****************************************************************/
+
+	//topoEventDelete := events.NewTopoEvent(device1.ID, events.EventItemDeleted, &device1)
```
Comment (Andrea-Campanella) [inline pkg/southbound/synchronizer/factory_test.go]: was this left out on purpose ? or is it useless ?

File: pkg/southbound/mocks/target_mock.go [hunk 2]
```diff
@@ -0,0 +1,112 @@
+// Code generated by protoc-gen-gogo. DO NOT EDIT.
```
Comment (ray-milkey) [inline pkg/southbound/mocks/target_mock.go]: Can we rename this file to target_mock_test.go? If we do that, the coverage tools will not count the mocked code in here against us.

File: pkg/southbound/synchronizer/factory_test.go (hunk 1 above)
Comment (ray-milkey) [inline pkg/southbound/synchronizer/factory_test.go]: Do we need this commented out portion?

Comment (SeanCondon) [reply pkg/southbound/synchronizer/factory_test.go]: leaving it out

Comment (SeanCondon) [reply pkg/southbound/synchronizer/factory_test.go]: Leaving it out

File: pkg/southbound/mocks/target_mock.go (hunk 2 above)
Comment (SeanCondon) [reply pkg/southbound/mocks/target_mock.go]: Actually I can't - if I make this _test.go then the main code can't find it.
Also the linter complains I have a test file but nothing to test it on. I'd have to create a source file with dummy code.

**Pull request merged**